cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -421,31 +404,17 @@ impl BranchTarget {
     /// byte of the target.  It does not take into account the Intel-specific
     /// rule that a branch offset is encoded as relative to the start of the
     /// following instruction.  That is a problem for the emitter to deal
-    /// with.
-    pub fn as_offset_i32(&self) -> Option<i32> {
+    /// with. If a label, returns zero.
+    pub fn as_offset32_or_zero(&self) -> i32 {
         match self {
-            &BranchTarget::ResolvedOffset(_, off) => {
+            &BranchTarget::ResolvedOffset(off) => {
                 // Leave a bit of slack so that the emitter is guaranteed to
                 // be able to add the length of the jump instruction encoding
                 // to this value and still have a value in signed-32 range.
-                if off >= -0x7FFF_FF00isize && off <= 0x7FFF_FF00isize {
-                    Some(off as i32)
-                } else {
-                    None
-                }
-            }
-            _ => None,
-        }
-    }
-
-    /// Map the block index given a transform map.
-    pub fn map(&mut self, block_index_map: &[BlockIndex]) {
-        match self {
-            &mut BranchTarget::Block(ref mut bix) => {
-                let n = block_index_map[*bix as usize];
-                *bix = n;
+                assert!(off >= -0x7FFF_FF00 && off <= 0x7FFF_FF00);
```

Comment:
This change concerns me somewhat.  What guarantees that this assertion can't fail now?

Reply:
Nothing guarantees it, but this form (`ResolvedOffset`) is used only when the lowering explicitly selects it, now; ordinary branches don't go through this code (the `LabelUse` handles them instead, and we can implement 64-bit long-form veneers there if we think we need >2GB code-size).